framework/src/utils/SlepcSupport.C

```diff
@@ -105,6 +109,23 @@ storeSlepcEigenProblemOptions(EigenProblem & eigen_problem, const InputParameter
   const std::string & which_eigen_pairs = params.get<MooseEnum>("which_eigen_pairs");
   eigen_problem.solverParams()._which_eigen_pairs =
       Moose::stringToEnum<Moose::WhichEigenPairs>(which_eigen_pairs);
+
+  // Set necessary parametrs used in EigenSystem::solve(),
+  // i.e. the number of requested eigenpairs nev and the number
+  // of basis vectors ncv used in the solution algorithm. Note that
+  // ncv >= nev must hold and ncv >= 2*nev is recommended
+  unsigned int n_eigen_pairs = params.get<unsigned int>("n_eigen_pairs");
+  unsigned int n_basis_vectors = params.get<unsigned int>("n_basis_vectors");
+
+  eigen_problem.setNEigenPairsRequired(n_eigen_pairs);
+
+  eigen_problem.es().parameters.set<unsigned int>("eigenpairs") = n_eigen_pairs;
+
+  // If the subspace dimension is too small, we increase it automatically
+  if (2 * n_eigen_pairs > n_basis_vectors)
+    n_basis_vectors = 2 * n_eigen_pairs;
```

Comment:
Can we issue a warning here saying that the user provided parameter has been overwritten by the code due to bla bla?